### PR TITLE
update dependencies

### LINKED ISSUE #234: @polkadot/* breaking change(s): Friday, November 27th
A `@polkadot` package has updated in the valid semver range with a breaking change.
This will cause newly created `@cennznet/api` projects to break on install.

**details:**  
internal/modules/cjs/loader.js:638 throw err; ^ Error: Cannot find module '@polkadot/rpc-core/rxjs' at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15) at Function.Module._load (internal/modules/cjs/loader.js:562:25) at Module.require (internal/modules/cjs/loader.js:692:17) at require (internal/modules/cjs/helpers.js:25:18) at Object.<anonymous> (/home/runner/work/api.js/api.js/new-project/node_modules/@cennznet/api/derives/attestation/getClaim.js:10:13) at Module._compile (internal/modules/cjs/loader.js:778:30) at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10) at Module.load (internal/modules/cjs/loader.js:653:32) at tryModuleLoad (internal/modules/cjs/loader.js:593:12) at Function.Module._load (internal/modules/cjs/loader.js:585:3)

### PR DESCRIPTION
**Pull request by KarishmaBothara**
updating dependencies to latest polkadot ones and fixing all the build issues
closes #234

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "pretty-quick --staged --pattern 'packages/**/src/**/*' && npx eslint . --ext .ts"
+      "pre-commit": "pretty-quick --staged --pattern 'packages/api/src/**/*' && npx eslint . --ext .ts"
```
Comment (KarishmaBothara) [inline package.json:63]: excluding types from pretty-quick as auto generated types now contain new syntax which is 
import type { Vec, u16, u32, u64 } from '@polkadot/types';
which makes prettier unhappy i,e
/Users/karishma/work/test/api.js/node_modules/prettier/index.js:9750
      throw error;
      ^

SyntaxError: '=' expected. (4:13)
  2 | /* eslint-disable */
  3 | 
> 4 | import type { u256 } from '@polkadot/types';
    |             ^
  5 |

File: scripts/generateSlimMetadata.ts [hunk 2]
```diff
@@ -1,85 +1,87 @@
-// Running this file with ~ yarn generate-slim-metadata 'wss://cennznet.unfrastructure.io/public/ws' will create slim metadata
```
Comment (KarishmaBothara) [inline scripts/generateSlimMetadata.ts:1]: Commented whole file.. need to fix this

File: scripts/MetadataMd.ts [hunk 3]
```diff
@@ -1,315 +1,315 @@
-// Copyright 2017-2020 @polkadot/typegen authors & contributors
```
Comment (KarishmaBothara) [inline scripts/MetadataMd.ts:1]: Commented this file... need to fix it

**Comment by KarishmaBothara** [comment]
Integration test are failing with the same reason as https://github.com/polkadot-js/api/issues/2877

File: package.json (hunk 1 above)
Comment (alexsednz) [reply package.json:63]: It makes sense to exclude the auto-generated code from such checks.

File: packages/types/src/interfaces/augment-api-query.ts [hunk 4]
```diff
@@ -100,13 +60,13 @@ declare module '@polkadot/api/types/storage' {
        **/
       uncles: AugmentedQuery<ApiType, () => Observable<Vec<UncleEntryItem>>> & QueryableStorageEntry<ApiType>;
     };
+    // @ts-ignore
     babe: {
```
Comment (KarishmaBothara) [inline packages/types/src/interfaces/augment-api-query.ts:64]: I don't know why it add babe here, when babe already exist.. investigating this.. add @ts-ignore for now

File: packages/types/src/interfaces/augment-api-tx.ts [hunk 5]
```diff
@@ -2061,89 +1579,73 @@ declare module '@polkadot/api/types/submittable' {
...
        * - Limited storage reads.
        * - One DB change.
        * # </weight>
        **/
-      unassignCurator: AugmentedSubmittable<
-        (bountyId: Compact<ProposalIndex> | AnyNumber | Uint8Array) => SubmittableExtrinsic<ApiType>
-      >;
+      unassignCurator: AugmentedSubmittable<(bountyId: Compact<ProposalIndex> | AnyNumber | Uint8Array) => SubmittableExtrinsic<ApiType>>;
     };
+    // @ts-ignore
```
Comment (KarishmaBothara) [inline packages/types/src/interfaces/augment-api-tx.ts:1607]: need to investigate the reason why auto-generate adds utility, when it already exist.. @ts-ignored for now

**Comment by KarishmaBothara** [comment]
> Integration test are failing with the same reason as [polkadot-js/api#2877](https://github.com/polkadot-js/api/issues/2877)
> seems 2.8.1 version is not stable..

https://github.com/polkadot-js/api/blob/master/CHANGELOG.md#281-nov-23-2020
Changelog says - Upgrade priority: Low. Recommended if using Substrate master with new MultiAddress.

**Review by holygits — COMMENTED** [review]
changes lgtm. what will be the side effects of ignoring those things from the augment files?

File: packages/types/src/interfaces/augment-api-query.ts (hunk 4 above)
Comment (KarishmaBothara) [reply packages/types/src/interfaces/augment-api-query.ts:64]: I think the reason of creation of this field is -
a. In node (CENNZnet) the state of Babe is little behind to what is available with substrate.. i.e authorVrfRandomness does not exist in ours..
b. Polkadot's latest version takes into account that..
For more information check attached file..
Screen Shot 2020-11-30 at 2 09 13 PM

File: packages/types/src/interfaces/augment-api-tx.ts (hunk 5 above)
Comment (KarishmaBothara) [reply packages/types/src/interfaces/augment-api-tx.ts:1607]: Similarly, the metadata information that comes from CENNZnet node for utility pallet is different from latest substrate..
it has additional function named - batchAll.. and some more weight related information..
Screen Shot 2020-11-30 at 2 14 12 PM

Comment (holygits) [reply packages/types/src/interfaces/augment-api-tx.ts:1607]: ah I see, do you know if these definitions will merge or overwrite each other?

Comment (KarishmaBothara) [reply packages/types/src/interfaces/augment-api-tx.ts:1607]: I think it overwrites. cause when I look at api.tx.batch object or even in UI.. I don't see batchAll

**Comment by alexsednz** [comment]
What is this PR waiting for?

**Comment by KarishmaBothara** [comment]
> What is this PR waiting for?

Merging can be performed automatically with 1 approving review.

File: packages/types/src/interfaces/augment-api-query.ts (hunk 4 above)
Comment (alexsednz) [reply packages/types/src/interfaces/augment-api-query.ts:64]: So what does ts-ignore doing here? what is ignored?

Comment (KarishmaBothara) [reply packages/types/src/interfaces/augment-api-query.ts:64]: We are ignore typescript error.. 
TS2717: Subsequent property declarations must have the same type. Property 'babe' must be of type '{ [key: string]: QueryableStorageEntry<ApiType>; authorities: MethodResult<ApiType, () => Observable<Vec<ITuple<[AuthorityId, BabeAuthorityWeight]>>>> & StorageEntryBase<...> & QueryableStorageEntry<...>; ... 10 more ...; underConstruction: MethodResult<...> & ... 1 more ... & QueryableStorageEntry<...>; }', but here has type '{ [key: string]: QueryableStorageEntry<ApiType>; authorities: MethodResult<ApiType, () => Observable<Vec<ITuple<[AuthorityId, BabeAuthorityWeight]>>>> & StorageEntryBase<...> & QueryableStorageEntry<...>; ... 9 more ...; underConstruction: MethodResult<...> & ... 1 more ... & QueryableStorageEntry<...>

Polkadot's type generation script uses Module Augmentation so when it has its own definition of Babe, it shouts with ts error

**Review by alexsednz — APPROVED** [review]

**Comment by alexsednz** [comment]
> > What is this PR waiting for?
> 
> Merging can be performed automatically with 1 approving review.

The title was suggesting that it's a WIP

File: packages/api/test/e2e/api.calls.e2e.ts [hunk 6]
```diff
@@ -66,9 +66,9 @@ describe('e2e api calls', () => {
     const provider = `wss://node-6711773975684325376.jm.onfinality.io/ws?apikey=${API_KEY}`;
```
Comment (holygits) [inline packages/api/test/e2e/api.calls.e2e.ts:66]: This test needs to use a specific block hash that will always be v36

**Review by holygits — CHANGES_REQUESTED** [review]

File: packages/api/test/e2e/api.calls.e2e.ts (hunk 6 above)
Comment (holygits) [reply packages/api/test/e2e/api.calls.e2e.ts:66]: meant to raise in the original PR where this test was added but forgot

Comment (KarishmaBothara) [reply packages/api/test/e2e/api.calls.e2e.ts:66]: done

**Review by holygits — APPROVED** [review]

**Pull request merged**